ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerview.js

```diff
@@ -73,7 +73,7 @@
 
         setModel(model) {
             super.setModel(model);
-            if (!this.#noFormats()) {
+            if (!this.#noFormats() && !model.getState().readOnly) {
```

Comment:
There are multiple issuse with this,
1. readOnly can be set dynamically, so please that use-case, when readOnly would be updated at runtime updateReadOnly would be called which should enable the datePicker popup. (https://opensource.adobe.com/aem-forms-af-runtime/js-docs/classes/Field.html#readonly
2. Please use the readOnly from the model, don't use state here.